util/rate_limiter_test.cc

```diff
@@ -89,6 +90,43 @@ TEST_F(RateLimiterTest, GetTotalRequests) {
          "Env::IO_TOTAL";
 }
 
+TEST_F(RateLimiterTest, GetTotalPendingRequests) {
+  std::unique_ptr<RateLimiter> limiter(
+      NewGenericRateLimiter(20 /* rate_bytes_per_sec */));
+  for (int i = Env::IO_LOW; i <= Env::IO_TOTAL; ++i) {
+    ASSERT_EQ(limiter->GetTotalPendingRequests(static_cast<Env::IOPriority>(i)),
+              0);
+  }
+
+  SyncPoint::GetInstance()->SetCallBack(
+      "GenericRateLimiter::Request:PostEnqueueRequest", [&](void* arg) {
+        port::Mutex* request_mutex = (port::Mutex*)arg;
+        // We temporarily unlock the mutex so that the following
+        // GetTotalPendingRequests() can acquire it
+        request_mutex->Unlock();
+        EXPECT_EQ(limiter->GetTotalPendingRequests(Env::IO_USER), 1);
```

Comment:
Good idea to check `GetTotalPendingRequests()` in the `Request()` thread so we don't have to worry about race conditions.

Maybe also assert the callback is called so minor changes, like initializing `RateLimiter` with `available_bytes_ = refill_bytes_per_period_`, won't defeat the purpose of the test. That could be done by (1) introducing a flag, `nonzero_pending_requests_verified_ = false` (or better name), before `EnableProcessing()`; (2) setting `nonzero_pending_requests_verified_ = true` in the callback; and (3) adding an `EXPECT_EQ` for `nonzero_pending_requests_verified_` to be true after calling `Request()`.

Reply:
Ahh - that's a good testing habit to make sure callback is called if there is any assertion done in it. I will update this PR and open a task to improve others in this test. Thanks!